stix2/properties.py

```diff
@@ -215,26 +212,15 @@ def clean(self, value):
 
 class TimestampProperty(Property):
 
-    def clean(self, value):
-        if isinstance(value, dt.date):
-            if hasattr(value, 'hour'):
-                return value
-            else:
-                # Add a time component
-                return dt.datetime.combine(value, dt.time(), tzinfo=pytz.utc)
+    def __init__(self, precision=None, **kwargs):
+        self.precision = precision
+        super(TimestampProperty, self).__init__(**kwargs)
 
-        # value isn't a date or datetime object so assume it's a string
+    def clean(self, value):
         try:
-            parsed = parser.parse(value)
-        except TypeError:
-            # Unknown format
-            raise ValueError("must be a datetime object, date object, or "
-                             "timestamp string in a recognizable format.")
-        if parsed.tzinfo:
-            return parsed.astimezone(pytz.utc)
-        else:
-            # Doesn't have timezone info in the string; assume UTC
-            return pytz.utc.localize(parsed)
+            return parse_into_datetime(value, self.precision)
+        except ValueError:
```

Comment:
I don't think we need the `try`/`except`/`raise` here (we aren't rewriting or adding any additional information to the Exception). Unless I'm missing something.